### PR TITLE
feat: Allow saving theme customizations

### LINKED ISSUE #11164: Allow saving theme in BE

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR adds API to save a customized theme so that it can be used again for that application. It also adds permission to themes. Each customized theme will have permissions set just like other domain objects.

Fixes #11164

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- JUnit test
- Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java [hunk 1]
```diff
@@ -4744,37 +4745,6 @@ public void migrateFirestorePluginToUqi3(MongockTemplate mongockTemplate) {
         mongockTemplate.save(firestorePlugin);
     }
 
-    @ChangeSet(order = "108", id = "create-system-themes", author = "")
```
[the remark this answers is not in the document]
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:4747]: Actually, the new migration is doing the same as the old one so I thought there is no need to keep them both. You are right, I should keep the old one also to preserve the consistency of the migration numbers.

[the remark this answers is not in the document]
Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java:4747]: Yes please. Use a different order number and suffix the `id` with a `-1` or `-repeat` or something, and in the function body, just call the other function. If it's the same thing, no need to repeat the whole code. Also, please include a comment on why this migration needs to be repeated. 😄

This was done before: https://github.com/appsmithorg/appsmith/blob/5022df91924c2cd7f4803c5c7ecd65a841ccf897/app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog.java#L4913-L4918

**Review by sharat87 — APPROVED** [review]

**Pull request merged**